core/src/main/kotlin/team/backend/config/ReactiveQueryConfig.kt

```diff
@@ -0,0 +1,110 @@
+        return SubqueryCreatorImpl()
+    }
+
+    @Bean
+    fun entityManagerFactory(@Qualifier("primaryDataSource") dataSource: DataSource): LocalContainerEntityManagerFactoryBean {
+        val emf = LocalContainerEntityManagerFactoryBean()
+        emf.dataSource = dataSource
+        emf.setPersistenceProviderClass(ReactivePersistenceProvider::class.java)
+        emf.setPackagesToScan("team.backend.domain.entity")
+        emf.jpaVendorAdapter = jpaVendorAdapters()
+        emf.setJpaProperties(jpaProperties())
+
+        return emf
+    }
+
+    @Bean
+    fun mutinySessionFactory(@Qualifier("primaryDataSource") dataSource: DataSource, localSessionFactoryBean: LocalContainerEntityManagerFactoryBean): Mutiny.SessionFactory {
+        val reactivePersistenceInfo = ReactivePersistenceInfo(
+            localSessionFactoryBean.persistenceUnitInfo!!,
+            localSessionFactoryBean.jpaPropertyMap,
+            dataSource
+        )
+
+        return ReactivePersistenceProvider()
+            .createContainerEntityManagerFactory(reactivePersistenceInfo, reactivePersistenceInfo.properties)
+            .unwrap(Mutiny.SessionFactory::class.java)
+    }
+
+    class ReactivePersistenceInfo(
+        persistenceUnitInfo: PersistenceUnitInfo,
+        jpaPropertyMap: Map<String, Any>,
+        dataSource: DataSource
+    ):
+        PersistenceUnitInfo by persistenceUnitInfo {
+
+        private val internalProps = Properties(persistenceUnitInfo.properties).apply {
+            putAll(jpaPropertyMap)
+            setProperty(Settings.SQL_CLIENT_POOL, MySQLConnectionPool::class.qualifiedName)
+            setProperty(Settings.SQL_CLIENT_POOL_CONFIG, VertxMySQLDBConnectionPoolConfiguration::class.qualifiedName)
+            setProperty(Settings.URL, (dataSource.unwrap(HikariDataSource::class.java) as HikariDataSource).jdbcUrl)
+            setProperty(Settings.USER, (dataSource.unwrap(HikariDataSource::class.java) as HikariDataSource).username)
+            setProperty(Settings.PASS, (dataSource.unwrap(HikariDataSource::class.java) as HikariDataSource).password)
+            setProperty(Settings.HBM2DDL_AUTO, "create")
+        }
+
+        override fun getProperties(): Properties = internalProps
+
+        override fun getPersistenceProviderClassName(): String = ReactivePersistenceProvider::class.qualifiedName!!
+    }
+
+    @Bean
+    fun queryFactory(
+        sessionFactory: Mutiny.SessionFactory,
+        subqueryCreator: SubqueryCreator
+    ): SpringDataHibernateMutinyReactiveQueryFactory {
+        return SpringDataHibernateMutinyReactiveQueryFactory(
+            sessionFactory = sessionFactory,
+            subqueryCreator = subqueryCreator
+        )
+    }
+
+    private fun jpaVendorAdapters(): JpaVendorAdapter = HibernateJpaVendorAdapter().apply {
+        setDatabasePlatform("org.hibernate.dialect.MySQLDialect")
+    }
+
+    private fun jpaProperties(): Properties = Properties().apply {
```

Comment:
Doesn't it conflict with the yml file?

Reply:
Yes. yml doesn't requirement. I was deleted 

- https://github.com/withDogecoin/chaterview-private/pull/2